tabs.js

```diff
@@ -0,0 +1,135 @@
+var h = require('hyperscript')
+var u = require('./util')
+  each = u.each,
+  find = u.find,
+  isVisible = u.isVisible,
+  setVisible = u.setVisible,
+  setInvisible = u.setInvisible
+
+function toggle_focus(page) {
+  isVisible(page)
+    ? blur(page)
+    : focus(page)
+}
+
+function focus(page) {
+  if (isVisible(page)) return
+
+  setVisible(page)
+  var el = page.firstChild
+  el.dispatchEvent(new CustomEvent('focus', {target: el}))
+}
+
+function blur (page) {
+  if (!isVisible(page)) return
+
+  setInvisible(page)
+  var el = page.firstChild
+  el.dispatchEvent(new CustomEvent('blur', {target: el}))
+}
+
+function moveTo(page, content, i) {
+  if(!content.children.length || i >= content.children.length)
+    content.appendChild(page)
+  else
+    content.insertBefore(page, content.children[i])
+}
+
+module.exports = function (content, onSelect) {
+  var tabs = h('section.tabs')
+  var selection
+
+  function build_tab (page) {
+    var link = h('a.link', {
+      href: '#',
+      onclick: function (ev) {
+        if(ev.shiftKey) toggle_focus(page)
+        else {
+          each(content.children, function (_page) {
+            if(_page == page) focus(_page)
+            else blur(_page)
+          })
+        }
+        ev.preventDefault()
+        ev.stopPropagation()
+      }},
+      getTitle(page)
+    )
+    var rm = h('a.close', {
+      href: '#',
+      onclick: function (ev) {
+        page.parentNode.removeChild(page)
+        tabs.removeChild(tab)
+      }},
+      'x'
+    )
+
+    var tab = h('div.tab', link, rm)
+
+    function updateTabClasses () {
+      if(isVisible(page))
+        tab.classList.add('-selected')
+      else
+        tab.classList.remove('-selected')
+
+      if(page.classList.contains('-notify'))
+        tab.classList.add('-notify')
+      else
+        tab.classList.remove('-notify')
+    }
+
+    function getTitle(page) {
+      var el = page.firstChild
+      return el.title || el.id || el.tagName
+    }
+
+    new MutationObserver(function (changes) {
+      if(page.title !== link.innerText)
+        link.innerText = getTitle(page) 
+      updateTabClasses()
+      onSelect && onSelect()
+    }).observe(page, {attributes: true, attributeFilter: ['title', 'style', 'class']})
+
+    new MutationObserver(function (changes) {
+      if(page.title !== link.innerText)
+        link.innerText = getTitle(page) 
+      updateTabClasses()
+      onSelect && onSelect()
+    }).observe(page.firstChild, {attributes: true, attributeFilter: ['title', 'style', 'class']})
```

Comment:
Might need some advice about what's best here @dominictarr : 

Given I've wrapped each provided `el` into a `div.page`, what should we listen to?
 - are these expensive ? 
 - how often are we using the changing of the `el.title` in order to update the tabs (formally menu)